src/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -3679,6 +3682,10 @@ public static boolean shallowEq(Object x, Object y) {
             if (y instanceof Boolean) {
                 return x.equals(y);
             }
+        } else if (x instanceof SymbolKey) {
```

Comment:
Don't  x.equals(y) and y.equals(x) yield the same result, so you can so `if (x instanceof symbolKey || y instanceof symbolKey)` ?

Reply:
Yes that is possible but then the code relies on the assumption that x == null. This is correct here but the null check is so many lines above that it might be overlooked some day.
For me this is the more robust and more readable (because it starts no questions in my mind when reading) version.

Reply:
k